README.md

```diff
@@ -78,103 +101,366 @@ Modify the app's **android/app/src/main/AndroidManifest.xml** to include the fol
+    identifier: 'identifier',
+    isAnonymous: false,
+    email: 'user_by_object@email.com',
+    firstName: 'first name',
+    fullName: 'full name',
+    uuid: 'uuid'
+}
+
+RaygunClient.setUser(user);
+
+// Alternatively
+RaygunClient.setUser("identifier");
+
+// Reset to ANON user
+RaygunClient.setUser(null);
+```
+
+<br/>
+<br/>
+
+###  clearSession() 
+Clears all information about the user and tags that have been set to this point.
+```typescript
+import RaygunClient from "raygun4reactnative"
+
+RaygunClient.clearSession();
+```
+
+<br/>
+<br/>
+
+###  recordBreadcrumb(message: string, details?: BreadcrumbOption) 
+Records Breadcrumbs that will be sent away with the CrashReportPayload.
+```typescript
+import RaygunClient, {BreadcrumbOption} from "raygun4reactnative"
+
+const bco: BreadcrumbOption = {
+    category: 'Some String you choose',
+    level: 'debug',
+    customData: {"Key_6": "My Data is bland"},
+}
+
+RaygunClient.recordBreadcrumb("Some message", bco);
+```
+
+<br/>
+<br/>
+
+###  sendError(error: Error, ...params: any) 
+Allows for an error to be sent to the Crash Reporting error handler along with some customized
+data. 'params' can be configured in the following ways:
+    1) data: CustomData, ... tags: string
+    2) data: CustomData
+    3) ... tags: string
+If custom data is being parsed with this function, ensure it is placed first before any tags. Also
+ensure that the custom data is a [CustomData](#customdata) instance, all tags will be strings.
+```typescript
+import RaygunClient, {CustomData} from "raygun4reactnative"
+
+const error = new Error("Custom Message"); // This error can be any Error
+const customData: CustomData = {"KEY_1": "VALUE_1"};
+
+// Example 1 (with Custom Data, with Tags)
+RaygunClient.sendError(error, customData, "tag1", "tag2", "tag3");
+
+// Example 2 (with Custom Data, WITHOUT Tags)
+RaygunClient.sendError(error, customData);
+
+// Example 3 (WITHOUT Custom Data, with Tags)
+RaygunClient.sendError(error, "tag1", "tag2", "tag3");
+```
+
+<br/>
+<br/>
+
+###  addCustomData(customData: CustomData) 
+Appends custom data to the current set of custom data.
+```typescript
+import RaygunClient, {CustomData} from "raygun4reactnative"
+
+const customData: CustomData = {"KEY_2": "VALUE_2"};
+
+RaygunClient.addCustomData(customData);
 ```
 
+<br/>
+<br/>
 
-## setUser(user: User | string)
\ No newline at end of file
+###  updateCustomData(updater: (customData: CustomData) => CustomData) 
+Apply some transformation lambda to all the custom data.
+
+```typescript
+import RaygunClient, {CustomData} from "raygun4reactnative"
+
+const updater = (customData: CustomData) => {
+  console.log(customData);
+  return customData;
+} 
+
+RaygunClient.addCustomData(updater);
+```
+
+<br/>
+<br/>
+
+###  sendRUMTimingEvent(eventType: RealUserMonitoringTimings, name: string, timeUsedInMs: number) 
+Construct a Real User Monitoring Timing Event and send it to the Real User Monitor to be transmitted.
```

Comment:
I think this should be from the viewpoint of a user, instead of the internals of the provider. So: "Manually sends a new timing event to Real User Monitoring. This can be a ...."